graphql_compiler/schema_generation/sqlalchemy/__init__.py

```diff
@@ -7,7 +7,9 @@
 
 
 def get_sqlalchemy_schema_info_from_specified_metadata(
-    vertex_name_to_table, direct_edges, dialect, class_to_field_type_overrides=None
+    vertex_name_to_table, direct_edges,
+    dialect, class_to_field_type_overrides=None,
+    is_mssql_2014=False
```

Comment:
I agree about factoring the flag into a separate PR, good idea.

I don't think it's necessarily the best idea to make the flag be as general as `supports_array_agg` because this code does very specific stuff that is only supported in MSSQL, namely XML PATH. In theory we could name the flag `is_mssql`, but we kind of have broken the correspondence between dialect and output due to the whole issue with the dialect field having been set to mssql the whole time. Not sure what the best solution is here.

Reply:
EDIT: I see that SQL SERVER also has XML PATH syntax. I will need to check whether my queries correctly execute on SQL Server 1.0 before making the flag more general.

Reply:
Yet another idea is passing a `dialect_specific_flags` dict argument, and validating that the arguments apply to the dialect specified.

Reply:
Broadly, I think we want two things here — h/t @lodrion for the fruitful discussion where these ideas crystallized:
- We want the flexibility of feature flags, so we can easily toggle between different implementations (e.g. specifically for `@fold`, we can choose to emit `array_agg` vs JSON-based tricks vs `XML PATH`).
- Simultaneously, we want users to be able to use the compiler with minimal configuration and getting-started cost. This includes not worrying about what values to use for all the flags.

I think in the long run, our architecture might need to look something like this:
- For a given query, we compute the set of things it requires the backend to support (since not all backends support all operations). This is stuff like `@fold` or `@recurse` or `@optional + nested traversals`.
- Each backend has a set of feature flags that could be simple booleans (`recurse_supported == True`) or a choice among many options (`fold_strategy == None` if not supported, or `"array_agg" / "xml_path" ...` otherwise). For a given query, we use this info to check that the backend in question has the ability to support the requested operations.
- We have feature flag presets for various database versions, which we allow users to use / amend / completely ignore and build their own. E.g. for Postgres 11, `fold_strategy` is set to `"array_agg"`, and for MSSQL 2014 it's set to `"xml_path"`. If users set a flag to an invalid value, the backend errors out — definitely advanced, use-with-care functionality that way.
- In the long term, we offer functionality to autodetect the database version and select the dialect automatically when possible (e.g. across different versions of SQL databases).

Reply:
Hi, the most recent push settles for the short-term solution of simply having postgres and mssql side by side with mssql being default and the desired target determined by the dialect field of the sql schema info. 

I like the idea of, in the long-run, for each choice point in the code, having a flag that toggles the behavior, and bundling those flags into specific dialects/versions on the basis of how those dialects set the flags.

Would you like to see something like that integrated into this PR, or deferred until we have a need for several more meaningful flags?

Reply:
Probably deferred is fine and we can stick with the dialect for the sake of this PR.